src/components/Sidebar.tsx

```diff
@@ -190,34 +190,41 @@ function Sidebar({
     }
   }
 
-  function getActiveId() {
-    const activeItem = items.find(({ url }) => url === activeUrl)
+  function getIdForUrl(items: SidebarItem[], url: string): string | null {
+    for (const item of items) {
+      if (item.url === url) return getId(item)
 
-    if (activeItem) return activeItem.url || activeItem.name
+      if (Array.isArray(item.items) && item.items.length > 0) {
+        const id = getIdForUrl(item.items, url)
 
-    const activeParentItem = items
-    .filter(({ items }) => Array.isArray(items) && items.length > 0)
-    .find(({ items }) => items.find(({ url }) => url === activeUrl))
-
-    if (!activeParentItem) return activeUrl
+        if (id) return id
+      }
+    }
 
-    if (collapsed && activeParentItem) return activeParentItem.url || activeParentItem.name
+    return null
+  }
 
-    const activeChildItem = activeParentItem.items.find(({ url }) => url === activeUrl)
+  function isDeployedWithActiveChild(item: SidebarItem) {
+    return activeId && item && Array.isArray(item.items) && item.items.some(x => getId(x) === activeId)
```

Comment:
couldn't this be `Array.isArray(item?.items)`?

Reply:
Good eye.